static/js/containers/DashboardPage.js

```diff
@@ -270,6 +314,51 @@ class DashboardPage extends React.Component {
     dispatch(setDocsInstructionsVisibility(bool));
   };
 
+  setCouponNotificationVisibility = bool => {
+    const { dispatch } = this.props;
+    dispatch(setCouponNotificationVisibility(bool));
+  };
+
+  setActiveCoupon = apiCoupon => {
+    const { dispatch } = this.props;
+    // convert `amount` from string to decimal
+    const coupon = R.evolve({amount: Decimal}, apiCoupon);
+    dispatch(setActiveCoupon(coupon));
+  };
+
+  renderCouponDialog() {
+    const {
+      programs,
+      ui,
+      coupons,
+    } = this.props;
+    if ( !coupons.activeCoupon ) {
+      return null;
+    }
+    let couponProgram = null, couponCourse = null;
+    if ( programs.availablePrograms ) {
```

Comment:
You should be able to skip this check, it will always be a list of some kind because the initial value is an empty list